cephfs/cephfs_test.go

```diff
@@ -10,28 +10,36 @@ import (
 )
 
 var (
-	CephMountTest string = "/tmp/ceph/mds/mnt/"
+	CephMountTest = "/tmp/ceph/mds/mnt/"
 )
 
-func TestCreateMount(t *testing.T) {
+// mountCeph, we will be reusing this function for almost every test
+// to mount the ceph volume.  Functions should call and use the
+// provided mount as the function also handles unmounting when done.
+func withCephMount(f func(*cephfs.MountInfo) error) error {
 	mount, err := cephfs.CreateMount()
-	assert.NoError(t, err)
-	assert.NotNil(t, mount)
-}
-
-func TestMountRoot(t *testing.T) {
-	mount, err := cephfs.CreateMount()
-	assert.NoError(t, err)
-	assert.NotNil(t, mount)
+	if err != nil {
```

Comment:
I think i'm missing something here. Why are we removing the asserts and returning?

Reply:
The purpose of this function is as a helper for testing.  Caller functions will use `withCephMount` to use the provided ceph mount handler, without having to handle the repetitiveness of opening, configuring, closing the ceph mount.  As such, this function is not a part of the testing framework. Assert iirc, takes in as the first parameter the testing pointer.  This function merely returns the error, and leaves to the caller, if it is a testing function, to assert on error (which i believe i do).